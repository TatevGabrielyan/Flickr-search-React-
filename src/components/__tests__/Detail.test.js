import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mockComponent } from '../../utils/testUtils';

import Detail from '../Detail';
import store from '../../config/mockStore';

jest.mock('react-image-lightbox', () => {
  return props => mockComponent('Lightbox', props);
});

describe('Detail component', () => {
  const component = (
    <Provider store={store}>
      <Detail />
    </Provider>
  );
  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matchs', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
