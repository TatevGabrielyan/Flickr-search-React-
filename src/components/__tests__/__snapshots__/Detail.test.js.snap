// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component Snapshot matchs 1`] = `
<code
  name="Lightbox"
  props={
    Object {
      "imageTitle": <React.Fragment>
        <a
          className="DetailImageTitle"
          href="https://www.flickr.com/photos/158309862@N07/45414777002/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="DetailImageTitleText"
          >
            Open
          </span>
        </a>
        <span
          className="DetailImageTitleOwner"
        >
          $
          that is a huge cat
           by $
          sivappa.technology
        </span>
      </React.Fragment>,
      "mainSrc": "https://farm2.staticflickr.com/1956/45414777002_21af9f1855_b.jpg",
      "nextSrc": "https://farm2.staticflickr.com/1948/30526186977_7dd1391bbd_b.jpg",
      "onCloseRequest": [Function],
      "onMoveNextRequest": [Function],
      "onMovePrevRequest": [Function],
      "prevSrc": "https://farm2.staticflickr.com/1927/44741364214_0f0a5195e9_b.jpg",
    }
  }
/>
`;
